recipe-server/client/control/actions/FilterActions.js

```diff
@@ -0,0 +1,129 @@
+import {
+  makeApiRequest,
+  recipesNeedFetch,
+  recipesReceived,
+  filtersReceived,
+} from 'control/actions/ControlActions';
+
+import {
+  getFilterParamString,
+} from 'control/selectors/FiltersSelector';
+
+import cloneArrayValues from 'client/utils/clone-array';
+import { capFirstLetter } from 'client/utils/string-man';
+
+const SET_FILTER = 'SET_FILTER';
+const SET_TEXT_FILTER = 'SET_TEXT_FILTER';
+const LOAD_FILTERS = 'LOAD_FILTERS';
+const RESET_FILTERS = 'RESET_FILTERS';
+
+function formatFilterOption(option) {
+  let label;
+  let value;
+
+  // string type = we were given just an option
+  // (no keyed value/label)
+  if (typeof option === 'string') {
+    value = label = option;
+  // else if we get an object, then we
+  // can extract the key/value props
+  } else if (typeof option === 'object') {
+    label = option.value;
+    value = option.key;
+  }
+
+  return {
+    value,
+    label,
+  };
+}
+
+
+/**
+ * Load list of possible filters from remote API.
+ * This is stored in the `filters` reducer, and
+ * later used to populate relevant RecipeFilters components.
+ */
+function loadFilters() {
+  return dispatch =>
+    dispatch(makeApiRequest('fetchFilters'))
+      .then(filters => {
+        if (!filters) {
+          return;
+        }
+        const newFilters = [];
+        // format each recipe
+        for (const group in filters) {
+          if (!filters.hasOwnProperty(group)) {
+            break;
+          }
+          const newGroup = {
+            value: group,
+            label: capFirstLetter(group),
```

Comment:
ah! sorry I missed that it was the group name. okay, it lives! maybe just rename it to something like `capitalizeFirst` and it should be good to go.